service/service.go

```diff
@@ -12,87 +14,79 @@ import (
 	"github.com/ONSdigital/go-ns/audit"
-	"github.com/ONSdigital/go-ns/clients/dataset"
-	"github.com/ONSdigital/go-ns/elasticsearch"
-	"github.com/ONSdigital/go-ns/healthcheck"
-	"github.com/ONSdigital/go-ns/kafka"
-	"github.com/ONSdigital/go-ns/log"
-	"github.com/pkg/errors"
+
+	"github.com/ONSdigital/log.go/log"
 )
 
 // Service represents the necessary config for dp-search-api
 type Service struct {
-	Auditor                   audit.AuditorService
-	AuthAPIURL                string
-	BindAddr                  string
-	DatasetAPIURL             string
-	DefaultMaxResults         int
-	Elasticsearch             api.Elasticsearcher
-	ElasticsearchURL          string
-	EnvMax                    int
-	HealthCheckInterval       time.Duration
-	HealthCheckTimeout        time.Duration
-	MaxRetries                int
-	OutputQueue               searchoutputqueue.Output
-	SearchAPIURL              string
-	SearchIndexProducer       kafka.Producer
-	ServiceAuthToken          string
-	Shutdown                  time.Duration
-	SignElasticsearchRequests bool
-	HasPrivateEndpoints       bool
+	Auditor                    audit.AuditorService
+	AuthAPIURL                 string
+	BindAddr                   string
+	DatasetAPIClient           api.DatasetAPIClient
+	DefaultMaxResults          int
+	Elasticsearch              api.Elasticsearcher
+	ElasticsearchURL           string
+	EnvMax                     int
+	HealthCheck                *healthcheck.HealthCheck
+	HealthCheckCriticalTimeout time.Duration
+	MaxRetries                 int
+	OutputQueue                searchoutputqueue.Output
+	SearchAPIURL               string
+	SearchIndexProducer        *kafka.Producer
+	ServiceAuthToken           string
+	Shutdown                   time.Duration
+	SignElasticsearchRequests  bool
+	HasPrivateEndpoints        bool
 }
 
 // Start handles consumption of events
-func (svc *Service) Start() {
+func (svc *Service) Start(ctx context.Context) {
 
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
 
 	apiErrors := make(chan error, 1)
 
-	datasetAPINoAuth := dataset.NewAPIClient(svc.DatasetAPIURL, "", "")
-	datasetAPI := dataset.NewAPIClient(svc.DatasetAPIURL, svc.ServiceAuthToken, "")
-
-	healthTicker := healthcheck.NewTicker(
-		svc.HealthCheckInterval,
-		datasetAPINoAuth,
-		elasticsearch.NewHealthCheckClient(svc.ElasticsearchURL, svc.SignElasticsearchRequests),
-	)
+	svc.HealthCheck.Start(ctx)
 
 	api.CreateSearchAPI(
+		ctx,
 		svc.SearchAPIURL,
 		svc.BindAddr,
 		svc.AuthAPIURL,
 		apiErrors,
 		&svc.OutputQueue,
-		datasetAPI,
-		datasetAPINoAuth,
+		svc.DatasetAPIClient,
+		svc.ServiceAuthToken,
 		svc.Elasticsearch,
 		svc.DefaultMaxResults,
 		svc.HasPrivateEndpoints,
 		svc.Auditor,
+		svc.HealthCheck,
 	)
 
 	// blocks until a fatal error occurs
 	select {
 	case err := <-apiErrors:
-		log.ErrorC("api error received", err, nil)
+		log.Event(ctx, "api error received", log.ERROR, log.Error(err))
```

Comment:
I think it was decided to not kill the services on errors, and instead have an external restart on a healthcheck failure. This would mean that here we should only wait for the signals channel, and not for the apiErrors. We can still log the errors:

```
...
// log errors from apiErrors chan
go func (){
    for {
        select {
        case err := <- apiErrors:
            log.Event(...)
        }
    }
}()

// blocks until a signal is received
<- signals
...
```